make-prs.py

```diff
@@ -0,0 +1,251 @@
+    if res.status_code != requests.codes.ok:
+        logger.error(f"Error getting diff from PR {pull_request_number}.")
+        exit(1)
+
+    with open(DIFF_FILE, "w") as f:
+        f.write(res.text)
+
+
+def clone(repo: str):
+    logger.info("Cloning repository ...")
+
+    shutil.rmtree(WORKING_DIR, ignore_errors=True)
+
+    clone_url = f"git@github.com:{repo}.git"
+    command = ["git", "clone", clone_url, WORKING_DIR]
+
+    result = subprocess.run(command, capture_output=True, text=True)
+    if result.returncode != 0:
+        logger.error(f"Error cloning reop {repo}: {result.stderr}")
+        exit(1)
+
+
+def create_new_branch_based_on(base: str) -> str:
+    logger.info("Creating branch ...")
+
+    head = f"{base}-update"
+
+    command = [
+        "git",
+        "checkout",
+        "-b",
+        head,
+        f"origin/{base}",
+    ]
+    result = subprocess.run(command, cwd=WORKING_DIR, capture_output=True, text=True)
+    if result.returncode != 0:
+        logger.error(f"Error creating new branch {head}: {result.stderr}")
+        exit(1)
+
+    return head
+
+
+def apply_patch() -> bool:
+    logger.info("Applying patch ...")
+
+    command = [
+        "git",
+        "apply",
+        "--3way",
+        "../diff.patch",
+    ]
+    result = subprocess.run(command, cwd=WORKING_DIR, capture_output=True, text=True)
+    if result.returncode != 0:
+        if "conflicts" in result.stderr:
+            return True
+        else:
+            logger.error(f"Error applying diff: {result.stderr}")
+            exit(1)
+
+
+def commit_diff(head: str):
+    logger.info("Pushing changes ...")
+
+    add = ["git", "add", "."]
+    subprocess.run(add, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    commit = ["git", "commit", "-m", "chore: merging diff"]
+    subprocess.run(commit, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    push = ["git", "push", "-f", "origin", head]
+    result = subprocess.run(push, cwd=WORKING_DIR, capture_output=True, text=True)
+
+    if result.returncode != 0:
+        logger.error(f"Error pushing changes to branch {head}: {result.stderr}")
+        exit(1)
+
+
+def create_pr(repo: str, base: str, head: str, conflict: bool):
+    logger.info("Creating PR ...")
+
+    title = f"chore: merging diff into branch {base}"
+    body = f"Automated change: merging diff into branch {base}"
+
+    if conflict:
+        title = f"{title} CONFLICTS!"
+        body = f"**Conflicts! Need human intervention!**\n\n{body}"
+
+    repo = gh_client.get_repo(repo)
+    pr = repo.create_pull(base=base, head=head, title=title, body=body)
+    if not pr:
+        logger.error("Error creating PR.")
+        exit(1)
+    logger.info(f"PR {pr.number} created.")
+
+
+def get_all_chapters_branches(repo: str) -> typing.List[str]:
+    return sorted(
+        [
+            b.name
+            for b in gh_client.get_repo(repo).get_branches()
+            if re.search(r"\d", b.name)
+        ]
+    )
```

Comment:
`sorted()` takes any iterable, including a generator, so you can drop the `[ ]` here. Also, formatting suggestion:

```python
branches = gh_client.get_repo(repo).get_branches()
return sorted(b.name for b in branches if re.search(r"\d", b.name))
```

Can we use a slightly tighter regex here, like startswith `\d+_`? That would be `re.match(r"\d+_", b.name)` I think.